pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/BigDecimalColumnPreIndexStatsCollector.java

```diff
@@ -112,8 +113,13 @@ public int getCardinality() {
   }
 
   @Override
-  public boolean hasNull() {
-    return false;
+  public boolean hasNulls() {
```

Comment:
This API is not really used. We should rely on whether null value vector exist and non-empty to identify whether a column contains null values

Reply:
Attempted to serialized whether dictionary has null value but found that dictionary serialization format cannot be changed easily.